core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/common/BavetStaticDataBuildHelper.java

```diff
@@ -0,0 +1,81 @@
+package ai.timefold.solver.core.impl.score.stream.bavet.common;
+
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import ai.timefold.solver.core.impl.bavet.NodeNetwork;
+import ai.timefold.solver.core.impl.bavet.common.AbstractNodeBuildHelper;
+import ai.timefold.solver.core.impl.bavet.common.BavetAbstractConstraintStream;
+import ai.timefold.solver.core.impl.bavet.common.TupleSourceRoot;
+import ai.timefold.solver.core.impl.bavet.common.tuple.AbstractTuple;
+import ai.timefold.solver.core.impl.bavet.common.tuple.RecordingTupleLifecycle;
+import ai.timefold.solver.core.impl.domain.variable.declarative.ConsistencyTracker;
+import ai.timefold.solver.core.impl.score.buildin.SimpleScoreDefinition;
+import ai.timefold.solver.core.impl.score.constraint.ConstraintMatchPolicy;
+import ai.timefold.solver.core.impl.score.stream.common.inliner.AbstractScoreInliner;
+
+public final class BavetStaticDataBuildHelper<Tuple_ extends AbstractTuple> {
+    private final NodeNetwork nodeNetwork;
+    private final RecordingTupleLifecycle<Tuple_> recordingTupleLifecycle;
+    private final Class<?>[] sourceClasses;
+
+    public <Solution_> BavetStaticDataBuildHelper(BavetAbstractConstraintStream<Solution_> staticConstraintStream) {
```

Comment:
Some of this logic is very similar with the other build helper.
Consider sharing code.